### PR TITLE
autoPatchelfHook: move multiline hook into a function

### PR DESCRIPTION
**Pull request by Qyriad**
In [NixOS/nixpkgs#290081 stdenv: log hooks as they run](https://github.com/NixOS/nixpkgs/pull/290081) it came to attention that autoPatchelfHook is one of if not the only hook in Nixpkgs that is a multiline string expression. Almost all hooks are functions, which guard with something like `if [ -z "${dontDoTheThing-}" ]; then ...` in the function, or single-line strings which include that guard inline and then call the real function, e.g. `if [ -z "${dontDoTheThing-} ]; then doTheThing; fi`.

This commit moves autoPatchelfHook to the former, which seems to be the most common style now.

## Description of changes

There should be no functional changes.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
	- Mostly. autoPatchelfHook is largely used for proprietary packages, some of which rely on `requireFile`; I skipped those, and `nixpkgs-review` supposedly queued a great deal many other packages for build but didn't actually build them. Regardless, all packages that were attempted built did build.
- [ ] (N/A) Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (N/A) (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (N/A) (Module updates) Added a release notes entry if the change is significant
  - [ ] (N/A) (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Qyriad** [comment]
cc @philiptaron, since you were interested in this.

File: pkgs/build-support/setup-hooks/auto-patchelf.sh [hunk 1]
```diff
@@ -99,11 +108,4 @@ autoPatchelf() {
 #
 # shellcheck disable=SC2016
 # (Expressions don't expand in single quotes, use double quotes for that.)
```
Comment (philiptaron) [inline pkgs/build-support/setup-hooks/auto-patchelf.sh:110]: This comment refers to code no longer in evidence.

File: pkgs/build-support/setup-hooks/auto-patchelf.sh [hunk 2]
```diff
@@ -88,6 +88,15 @@ autoPatchelf() {
...
+        autoPatchelf -- $(for output in $(getAllOutputNames); do
+            [ -e "${!output}" ] || continue
+            echo "${!output}"
+        done)
+    fi
+}
+
 # XXX: This should ultimately use fixupOutputHooks but we currently don't have
 # a way to enforce the order. If we have $runtimeDependencies set, the setup
 # hook of patchelf is going to ruin everything and strip out those additional
```
Comment (philiptaron) [inline pkgs/build-support/setup-hooks/auto-patchelf.sh:102]: This whole comment belongs on the function, I think.

File: pkgs/build-support/setup-hooks/auto-patchelf.sh [hunk 3]
```diff
@@ -88,6 +88,15 @@ autoPatchelf() {
         --extra-args "${patchelfFlagsArray[@]}"
 }
 
+autoPatchelfEachOutput() {
```
Comment (philiptaron) [inline pkgs/build-support/setup-hooks/auto-patchelf.sh:91]: Naming nit: based on the comments below, this ought to be named `autoPatchelfPostFixup`

**Review by philiptaron — COMMENTED** [review]
Couple of naming and comment-related things to consider. I'm inclined to accept if they're explained or addressed.

File: pkgs/build-support/setup-hooks/auto-patchelf.sh (hunk 1 above)
Comment (Qyriad) [reply pkgs/build-support/setup-hooks/auto-patchelf.sh:110]: Good catch!

**Comment by Qyriad** [comment]
Thanks @philiptaron; fixed those up now!

**Comment by layus** [comment]
Changing the PR base went smoothly. This is a trivial change (but a mass rebuild nonetheless). Merging.
Thanks for your work :-)

**Pull request merged**